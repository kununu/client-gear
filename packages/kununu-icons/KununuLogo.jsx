import React from 'react';
import PropTypes from 'prop-types';

const Icon = ({className, ariaHidden}) => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 781.2 184.3"
    className={className}
    aria-hidden={ariaHidden}
  >
    <g fill="#fff">
      <path d="M104.6,121.9c0-12.1-4.7-23.1-12.4-31.5c7.5-8.3,12.1-19.2,12.1-31.1v-7.9h-0.1c-0.3-1.8-2-3.2-3.9-3.2    h-24c-2,0-3.6,1.4-3.9,3.2l-0.5,7.9c0,8.5-7.1,15.4-15.8,15.4v0.1h-8.1c-5.6,0-10.9,1-15.9,2.7V8.9C32,6.8,30.2,5,28,5H4    C1.8,5,0,6.8,0,8.9v145.3c0,2.1,1.8,3.9,4,3.9h24c2.2,0,4-1.8,4-3.9v-32.3h0.2c0-8.5,7.1-15.4,15.8-15.4h8.5    c8.7,0,15.8,6.9,15.8,15.4v7.9h0.1v24.4c0,2.1,1.8,3.9,4,3.9h24c2.2,0,4-1.8,4-3.9l0.4-24.4V121.9z" />
      <path d="M224.4,78.8c0-2.1-1.8-3.9-4-3.9h-24c-2.2,0-4,1.8-4,3.9v58.5c0,8.5-7.2,15.4-16,15.4H168    c-8.7,0-15.8-6.9-15.8-15.4l-0.1-58.5c0-2.1-1.8-3.9-4-3.9h-24c-2.2,0-4,1.8-4,3.9l-0.1,58.5c0,26,21.3,47.1,47.9,47.1h8.5    c26.6,0,48-21.1,48-47.1L224.4,78.8z" />
      <path d="M240.6,155.1c0,2.1,1.8,3.9,4,3.9h24c2.2,0,4-1.8,4-3.9l0-58.5c0-8.5,7.2-15.4,15.9-15.4h8.5    c8.7,0,15.8,6.9,15.8,15.4l0.1,58.5c0,2.1,1.8,3.9,4,3.9h24c2.2,0,4-1.8,4-3.9l0.1-58.5c0-25.9-21.3-47.1-47.9-47.1h-8.5    c-26.6,0-47.9,21.1-47.9,47.1L240.6,155.1z" />
      <path d="M465.4,78.8c0-2.1-1.8-3.9-4-3.9h-24c-2.2,0-4,1.8-4,3.9v58.5c0,8.5-7.2,15.4-15.9,15.4H409    c-8.7,0-15.8-6.9-15.8-15.4l-0.1-58.5c0-2.1-1.8-3.9-4-3.9h-24c-2.2,0-4,1.8-4,3.9l0,58.5c0,26,21.3,47.1,47.9,47.1h8.5    c26.6,0,47.9-21.1,47.9-47.1L465.4,78.8z" />
      <path d="M706.5,78.8c0-2.1-1.8-3.9-4-3.9h-24c-2.2,0-4,1.8-4,3.9v58.5c0,8.5-7.2,15.4-15.9,15.4h-8.5    c-8.7,0-15.8-6.9-15.8-15.4l-0.1-58.5c0-2.1-1.8-3.9-4-3.9h-24c-2.2,0-4,1.8-4,3.9l-0.1,58.5c0,26,21.3,47.1,47.9,47.1h8.5    c26.6,0,47.9-21.1,47.9-47.1L706.5,78.8z" />
      <path d="M481.7,154.2c0,2.2,1.8,3.9,4,3.9h24c2.2,0,4-1.7,4-3.9V95.8c0-8.5,7.2-15.4,15.9-15.4h8.5    c8.7,0,15.8,6.9,15.8,15.4l0.1,58.5c0,2.2,1.8,3.9,4,3.9h24c2.2,0,4-1.7,4-3.9l0.1-58.5c0-26-21.3-47.1-47.9-47.1h-8.5    c-26.6,0-47.9,21.1-47.9,47.1L481.7,154.2z" />
      <path d="M725.5,45.2" />
      <path d="M725.5,32.7" />
      <g>
        <path d="M727.6,13.9c-1.6-1.6-4.3-1.6-5.9,0l-1.6,1.5c-1.6,1.6-1.6,4.2,0,5.8l9.6,9.4c1.3,1.3,1.2,3.4-0.1,4.6     l-1.2,1.2c-1.3,1.3-3.4,1.3-4.7,0l-9.7-9.4c-5-4.8-5-12.7-0.1-17.6l1.6-1.5c5-4.8,13-4.8,18,0.1l9,8.8l0.6,0.6     c1.3,1.3,1.3,3.3,0,4.6l-1.2,1.2c-1.3,1.3-3.5,1.4-4.8,0.1L727.6,13.9z" />
        <path d="M771.5,21.3c1.6-1.6,1.6-4.2,0-5.7l-1.6-1.6c-1.6-1.6-4.3-1.6-5.9,0l-9.6,9.4c-1.3,1.3-3.4,1.2-4.7-0.1     l-1.2-1.2c-1.3-1.3-1.3-3.3-0.1-4.6l9.6-9.5c5-4.8,13-4.9,18-0.1l1.6,1.5c5,4.9,4.9,12.7,0,17.6l-9,8.8l-0.6,0.6     c-1.3,1.3-3.4,1.3-4.7,0l-1.2-1.2c-1.3-1.3-1.4-3.4-0.1-4.6L771.5,21.3z" />
        <path d="M720.1,56.9c-1.6,1.6-1.6,4.2,0,5.8l1.6,1.6c1.6,1.6,4.3,1.6,5.9,0l9.6-9.4c1.3-1.3,3.4-1.2,4.7,0.1     l1.2,1.2c1.3,1.3,1.3,3.3,0.1,4.6l-9.7,9.4c-5,4.9-13,4.9-18,0.1l-1.6-1.5c-5-4.8-4.9-12.7,0.1-17.5l9-8.8l0.6-0.6     c1.3-1.3,3.4-1.3,4.7,0l1.2,1.2c1.3,1.3,1.4,3.4,0.1,4.6L720.1,56.9z" />
        <path d="M763.9,64.1c1.6,1.6,4.3,1.6,5.9,0l1.6-1.5c1.6-1.6,1.7-4.2,0-5.8l-9.6-9.4c-1.3-1.3-1.2-3.3,0.1-4.6     l1.2-1.2c1.3-1.3,3.4-1.3,4.7-0.1l9.7,9.4c5,4.8,5,12.7,0.1,17.6l-1.6,1.5c-5,4.9-13,4.8-18-0.1l-9-8.8l-0.6-0.6     c-1.3-1.3-1.3-3.3,0-4.6l1.2-1.2c1.3-1.3,3.5-1.4,4.8-0.1L763.9,64.1z" />
      </g>
    </g>
  </svg>
);

Icon.propTypes = {
  ariaHidden: PropTypes.bool,
  className: PropTypes.string,
};

Icon.defaultProps = {
  ariaHidden: undefined,
  className: undefined,
};

export default Icon;
