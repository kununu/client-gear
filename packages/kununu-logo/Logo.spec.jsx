import React from 'react';
import {render} from 'enzyme'; // eslint-disable-line import/no-extraneous-dependencies

import KununuLogo from './index';

test('Renders Logo without crashing', () => {
  const component = render(
    <KununuLogo
      title="test"
    />,
  );

  expect(component).toMatchSnapshot();
});

test('Renders Logo with link', () => {
  const component = render(
    <KununuLogo
      title="test"
      link={<a href="/">Test</a>}
    />,
  );

  expect(component).toMatchSnapshot();
});
