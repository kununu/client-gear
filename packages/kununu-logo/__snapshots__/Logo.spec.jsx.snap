// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Logo with link 1`] = `
<a
  href="/"
>
  <picture>
    <source
      media="(max-width: 829px)"
      srcset="https://assets.kununu.com/images/footer/logo-mobile.svg"
    />
    <source
      media="(min-width: 830px)"
      srcset="https://assets.kununu.com/images/footer/logo-desktop.svg"
    />
    <img
      alt="test"
      aria-hidden="true"
      class="logo"
      rel="presentation"
      src="https://assets.kununu.com/images/footer/logo-desktop.svg"
    />
  </picture>
</a>
`;

exports[`Renders Logo without crashing 1`] = `
<picture>
  <source
    media="(max-width: 829px)"
    srcset="https://assets.kununu.com/images/footer/logo-mobile.svg"
  />
  <source
    media="(min-width: 830px)"
    srcset="https://assets.kununu.com/images/footer/logo-desktop.svg"
  />
  <img
    alt="test"
    aria-hidden="true"
    class="logo"
    rel="presentation"
    src="https://assets.kununu.com/images/footer/logo-desktop.svg"
  />
</picture>
`;
